### PR TITLE
Alerting: Refactor createUrl and createAbsoluteUrl 

### PR DESCRIPTION
**Pull request by soniaAguilarPeiron**
**What is this feature?**

- This PR renames `createUrl` to `createRelativeUrl`
- creates new type : `RelativeUrl` : `type RelativeUrl = `/${string}``
- updates the parameter in `createRelativeUrl`, and `createAbsoluteUrl` to be `RelativeUrl`
- It also add tests for both `createRelativeUrl`, and `createAbsoluteUrl`

This way we protect this parameter to start always with an slash

Fixes #

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: public/app/features/alerting/unified/utils/url.ts [hunk 1]
```diff
@@ -1,10 +1,14 @@
...
+) {
   const searchParams = new URLSearchParams(queryParams);
   const searchParamsString = searchParams.toString();
+  const pathWithSlash = path.startsWith('/') ? path : `/${path}`;
 
-  return `${config.appSubUrl}${path}${searchParamsString ? `?${searchParamsString}` : ''}`;
+  return `${config.appSubUrl}${pathWithSlash}${searchParamsString ? `?${searchParamsString}` : ''}`;
 }
 
 export function createAbsoluteUrl(
```
Comment (gillesdemey) [inline public/app/features/alerting/unified/utils/url.ts:14]: Any particular reason why we aren't also checking for leading `/` when passing path in to `createAbsoluteUrl`?

Comment (soniaAguilarPeiron) [reply public/app/features/alerting/unified/utils/url.ts:14]: no, let's do it!

File: public/app/features/alerting/unified/utils/url.ts [hunk 2]
```diff
@@ -1,10 +1,14 @@
 import { config } from '@grafana/runtime';
 
-export function createUrl(path: string, queryParams?: string[][] | Record<string, string> | string | URLSearchParams) {
+export function createRelativeUrl(
+  path: string,
```
Comment (konrad147) [inline public/app/features/alerting/unified/utils/url.ts:4]: Just an idea, what if we used templated type like `path: `/${string}` instead? It should force the function's users to always provide a path with a leading slash. I see the function automatically adds the leading slash now, but adding a slash gives you a slightly different result so maybe it's better to force the slash using types?

Comment (soniaAguilarPeiron) [reply public/app/features/alerting/unified/utils/url.ts:4]: I love this approach ❤️

Comment (soniaAguilarPeiron) [reply public/app/features/alerting/unified/utils/url.ts]: Added a commit with this approach, there is only [one place where typescript still complains](https://github.com/grafana/grafana/pull/90667/files#diff-0979422fa60152db34cd8fd05b18557ef04ccae254bf59153deb6da6a168e08dR55), but I think we are ok ignoring it

File: public/app/features/alerting/unified/components/rule-viewer/RuleViewer.tsx [hunk 3]
```diff
@@ -222,7 +222,7 @@ const createMetadata = (rule: CombinedRule): PageInfoItem[] => {
 // TODO move somewhere else
 export const createListFilterLink = (values: Array<[string, string]>) => {
   const params = new URLSearchParams([['search', values.map(([key, value]) => `${key}:"${value}"`).join(' ')]]);
-  return createRelativeUrl(`/alerting/list?` + params.toString());
+  return createRelativeUrl(`/alerting/list?${params.toString()}`);
```
Comment (konrad147) [inline public/app/features/alerting/unified/components/rule-viewer/RuleViewer.tsx:225]: nit: I think we could simplify to

```
return createRelativeUrl('/alerting/list', params)
```

**Review by konrad147 — APPROVED** [review]
LGTM!

**Review by gillesdemey — APPROVED** [review]

**Pull request merged**